insights/parsers/tests/test_engine_config.py

```diff
@@ -0,0 +1,296 @@
+SANWipeAfterDelete: false version: general
+SearchResultsLimit: 100 version: general
+ServerRebootTimeout: 300 version: general
+ConsoleReleaseCursorKeys: shift+f12 version: general
+ConsoleToggleFullScreenKeys: shift+f11 version: general
+SpiceUsbAutoShare: true version: general
+FullScreenWebadminDefault: false version: general
+FullScreenUserportalBasicDefault: true version: general
+FullScreenUserportalExtendedDefault: false version: general
+SpmCommandFailOverRetries: 3 version: general
+SPMFailOverAttempts: 3 version: general
+SpmVCpuConsumption: 1 version: general
+SSHInactivityTimeoutSeconds: 300 version: general
+SSHInactivityHardTimeoutSeconds: 1800 version: general
+NumberOfUSBSlots: 4 version: general
+SSLEnabled: true version: general
+StorageDomainFailureTimeoutInMinutes: 5 version: general
+StoragePoolRefreshTimeInSeconds: 10 version: general
+TimeoutToResetVdsInSeconds: 60 version: general
+DelayResetForSpmInSeconds: 20 version: general
+DelayResetPerVmInSeconds: 0.5 version: general
+TimeToReduceFailedRunOnVdsInMinutes: 30 version: general
+UserDefinedVMProperties:  version: 3.6
+UserDefinedVMProperties:  version: 4.0
+UserDefinedVMProperties:  version: 4.1
+UtilizationThresholdInPercent: 80 version: general
+ValidNumOfMonitors: 1,2,4 version: general
+VdcVersion: 4.1.0.0 version: general
+VDSAttemptsToResetCount: 2 version: general
+VdsLoadBalancingIntervalInMinutes: 1 version: general
+VdsRecoveryTimeoutInMinutes: 3 version: general
+VdsRefreshRate: 3 version: general
+vdsTimeout: 180 version: general
+vdsConnectionTimeout: 20 version: general
+vdsRetries: 0 version: general
+EventProcessingPoolSize: 10 version: general
+VmGracefulShutdownMessage: System Administrator has initiated shutdown of this Virtual Machine. Virtual Machine is shutting down. version: general
+VmGracefulShutdownTimeout: 30 version: general
+VM32BitMaxMemorySizeInMB: 20480 version: general
+VM64BitMaxMemorySizeInMB: 4194304 version: 3.6
+VM64BitMaxMemorySizeInMB: 4194304 version: 4.0
+VM64BitMaxMemorySizeInMB: 4194304 version: 4.1
+MaxMemorySlots: 16 version: general
+HotPlugMemoryMultiplicationSizeMb: 256 version: general
+VncKeyboardLayout: en-us version: general
+WaitForVdsInitInSec: 60 version: general
+FenceQuietTimeBetweenOperationsInSec: 180 version: general
+FenceProxyDefaultPreferences: cluster,dc version: general
+MaxAuditLogMessageLength: 10000 version: general
+SysPrepDefaultUser:  version: general
+SysPrepDefaultPassword: Empty version: general
+UserSessionTimeOutInterval: 30 version: general
+IPTablesConfig:
+# oVirt default firewall configuration. Automatically generated by vdsm bootstrap script.
+*filter
+:INPUT ACCEPT [0:0]
+:FORWARD ACCEPT [0:0]
+:OUTPUT ACCEPT [0:0]
+-A INPUT -m state --state ESTABLISHED,RELATED -j ACCEPT
+-A INPUT -p icmp -j ACCEPT
+-A INPUT -i lo -j ACCEPT
+# vdsm
+-A INPUT -p tcp --dport @VDSM_PORT@ -j ACCEPT
+# ovirt-imageio-daemon
+-A INPUT -p tcp --dport 54322 -j ACCEPT
```

Comment:
This current parsing logic is good enough for Line-122 type parameters.
Well, for this 'IPTablesConfig' block, it will go wrong in parsing.
If we don't have a better way to solve this, maybe it will be better to add some explanation in docstrings to notice the users pay attention to this.
@xiangce  Any ideas about this?

Reply:
Hi @JoySnow Yeah! that the a part which is not been parsed at-the-moment. I even added an assertion to test `IPTablesConfig`. Let me think over it a bit else I'll add appropriate explanation.

Reply:
@JoySnow 

I tried few regex patterns(they can be any multi-line command output in future) and they don't seems reliable. As for now I'll skip this kind of blocks and update the docs so that the user is informed.